### DISCUSSION TITLE
nucleo-144 f767zi with SSD1963(BD,800x480) 8bit parallel

### DISCUSSION THREAD

**Discussion by Karbonforms** [Bodmer/TFT_eSPI#994] [Q&A - General]
Is this possible with this library?

I have:
`#define STM32
#define STM_PORTA_DATA_BUS
#define TFT_PARALLEL_8_BIT
#define SSD1963_800_DRIVER
#define TFT_CS   PC8 
#define TFT_DC   PC9
#define TFT_RST  PC12

#define TFT_WR    PC10
#define TFT_RD    PC11

#define TFT_D0 PA0 // 8 bit Port A parallel bus to TFT
#define TFT_D1 PA1
#define TFT_D2 PA2
#define TFT_D3 PA3
#define TFT_D4 PA4
#define TFT_D5 PA5
#define TFT_D6 PA6
#define TFT_D7 PA7
`

Attempting to compile gives:
`In file included from .pio\libdeps\nucleo_f767zi\TFT_eSPI\TFT_eSPI.h:56,
                 from .pio\libdeps\nucleo_f767zi\TFT_eSPI\TFT_eSPI.cpp:17:
.pio\libdeps\nucleo_f767zi\TFT_eSPI\Processors/TFT_eSPI_STM32.c: In member function 'void TFT_eSPI::pushBlock(uint16_t, uint32_t)':
.pio\libdeps\nucleo_f767zi\TFT_eSPI\Processors/TFT_eSPI_STM32.h:732:38: error: redeclaration of 'uint8_t r'
  732 |     #define tft_Write_16(C)  uint8_t r = (((C) & 0xF800)>> 8); uint8_t g = (((C) & 0x07E0)>> 3); uint8_t b = (((C) & 0x001F)<< 3); \
      |                                      ^
.pio\libdeps\nucleo_f767zi\TFT_eSPI\Processors/TFT_eSPI_STM32.c:100:28: note: in expansion of macro 'tft_Write_16'
  100 |       tft_Write_16(color); tft_Write_16(color); tft_Write_16(color); tft_Write_16(color);
      |                            ^~~~~~~~~~~~
.pio\libdeps\nucleo_f767zi\TFT_eSPI\Processors/TFT_eSPI_STM32.h:732:38: note: 'uint8_t r' previously declared here
  732 |     #define tft_Write_16(C)  uint8_t r = (((C) & 0xF800)>> 8); uint8_t g = (((C) & 0x07E0)>> 3); uint8_t b = (((C) & 0x001F)<< 3); \`

With those last 3 ("redeclaration",  "expansion" and "previously declared here") repeated about eleventy billion times. Curiously, the redeclaration and previously-declared-here point to the same 'uint8_t r'

In my IDE(clion) uint8_t appears to be undefined, but slapping in a few #include <inttypes.h> still gives above errors. IDE just likely confused.

Cheers!

**Comment by Bodmer** [comment]
Yes, it will work.

For the STM32 the library expects the official [STM32 Arduino board package](https://github.com/stm32duino/Arduino_Core_STM32) to be loaded. Given the large number of error messages I think this is not visible to the PIO IDE.
